### PR TITLE
chore: refactor code to eliminate redundancy

### PR DESCRIPTION
**Pull request by jamesgeorge007**
**What kind of change does this PR introduce?**

refactoring

**Did you add tests for your changes?**
Nope

**If relevant, did you update the documentation?**
N/A
**Summary**
Made a minor refactor so as to eliminate redundancy.

N/A
**Does this PR introduce a breaking change?**

Nope
**Other information**
N/A

### CODE REVIEW

**Comment by jamesgeorge007** [comment]
cc @evenstensberg

**Comment by jamesgeorge007** [comment]
@evenstensberg `npm` gets into action where `yarn` was expected. I'm not sure whether if that is an issue concerning the present logic or the test case :thinking: 
What would you suggest?

**Comment by jamesgeorge007** [comment]
@rishabh3112 Thoughts :thinking:

**Comment by jamesgeorge007** [comment]
@ematipico Thoughts :thinking:

**Comment by evenstensberg** [comment]
This change is good

**Comment by jamesgeorge007** [comment]
@evenstensberg What about the build failure?

**Comment by ematipico** [comment]
@jamesgeorge007 your change broke some test. This means that your new logic has some flaw or the tests have to be revisited. Check the log of the CI and make sure all the tests are passing on your machine.

File: packages/utils/package-manager.ts [hunk 1]
```diff
@@ -66,6 +26,30 @@ export function getPackageManager(): string {
...
+ *
+ * Spawns a new process using the respective package manager
+ *
+ * @param {String} pkg - The dependency to be installed
+ * @param {Boolean} isNew - indicates if it needs to be updated or installed
+ * @returns {Function} spawn - Installs the package
+ */
+
+function spawnWithArg(pkg: string, isNew: boolean): SpawnSyncReturns<Buffer> {
+	const packageManager: string = getPackageManager();
```
Comment (jamesgeorge007) [inline packages/utils/package-manager.ts]: @ematipico It seems that the `getPackageManager()` method has some ambiguous logic which breaks those tests :thinking:

Comment (jamesgeorge007) [reply packages/utils/package-manager.ts]: https://github.com/webpack/webpack-cli/blob/6170fd0197b0aea8e00d3dc397ed754bc043d299/packages/utils/package-manager.ts#L15

Comment (evenstensberg) [reply packages/utils/package-manager.ts]: You can mock the same way, I agree. You need to revise nonetheless

Comment (jamesgeorge007) [reply packages/utils/package-manager.ts]: @evenstensberg How to proceed further?

Comment (jamesgeorge007) [reply packages/utils/package-manager.ts]: @evenstensberg The following test script for detecting yarn as the package manager fails each and every time.
https://github.com/webpack/webpack-cli/blob/6170fd0197b0aea8e00d3dc397ed754bc043d299/packages/utils/__tests__/package-manager.test.ts#L38

Comment (jamesgeorge007) [reply packages/utils/package-manager.ts]: @ematipico Thoughts :thinking:

Comment (jamesgeorge007) [reply packages/utils/package-manager.ts]: @evenstensberg The following test script for detecting yarn as the package manager fails each and every time.
https://github.com/webpack/webpack-cli/blob/6170fd0197b0aea8e00d3dc397ed754bc043d299/packages/utils/__tests__/package-manager.test.ts#L38

Comment (jamesgeorge007) [reply packages/utils/package-manager.ts]: @misterdev Thoughts :thinking:

**Comment by jamesgeorge007** [comment]
@evenstensberg The following test script for detecting yarn as the package manager fails each and every time.
https://github.com/webpack/webpack-cli/blob/6170fd0197b0aea8e00d3dc397ed754bc043d299/packages/utils/__tests__/package-manager.test.ts#L38

**Comment by evenstensberg** [comment]
Could you fix those or as a last resort disable them?

**Comment by jamesgeorge007** [comment]
@evenstensberg recently I rebased my branch from `next` which results in the tests to fail. I guess merging this to the `next` branch wouldn't be an issue as they're to be tracked on `master` later.

**Comment by jamesgeorge007** [comment]
@evenstensberg recently I rebased my branch from `next` which resulted in the tests to fail. I guess merging this to the `next` branch wouldn't be an issue as they're to be tracked on `master` later.

**Comment by evenstensberg** [comment]
Aaah yes. We should disable those tests for now. Could you be kind to open a PR that disables them before we merge this one?

**Comment by jamesgeorge007** [comment]
@evenstensberg So what you intend is to open yet another PR with 72cfdfa :thinking: 
Am I required to target `master` in that case?

**Comment by evenstensberg** [comment]
You can do `it.skip` I think, and no, you'll need to do it against this branch. This is the default branch

**Comment by jamesgeorge007** [comment]
@evenstensberg I guess you're expecting any new PR's once all the tests are passing on the `next` branch right?

**Comment by evenstensberg** [comment]
I changed a branch to a new next clone. Could you reopen? I'm currently updating tests there now

**Comment by jamesgeorge007** [comment]
I guess only those guys with administrative privileges can reopen the PR :thinking: 
Also, could you elaborate further, didn't get you :confused:

**Comment by jamesgeorge007** [comment]
@evenstensberg ??

**Comment by evenstensberg** [comment]
Gotta fix the src before we can review the PR's, reopening but will not review yet.

**Comment by jamesgeorge007** [comment]
@evenstensberg The branch was all messed up. Fixed it as per now :+1: 
Can you reopen?

**Comment by jamesgeorge007** [comment]
@evenstensberg The branch was all messed up. Fixed it as per now :+1: 
Can you reopen?

**Comment by jamesgeorge007** [comment]
Check out #1002

**Pull request closed without merging**